joinmarket/support.py

```diff
@@ -398,3 +410,61 @@ def debug_dump_object(obj, skip_fields=None):
             log.debug(pprint.pformat(v))
         else:
             log.debug(str(v))
+    
+def save_session_to_file (tumbler,wallet_file,destaddrs,filename):
+    if not os.path.exists(os.path.dirname(filename)):
+        try:
+            os.makedirs(os.path.dirname(filename))
+        except OSError as exc:
+            if exc.errno != errno.EEXIST:
+                raise
+    f = open(filename,"w")
+    f.write('{"options":')
+    f.write(json.dumps(tumbler.options)+',\n')
+    f.write('"wallet": "'+wallet_file+'",\n')
+    f.write('"destaddrs":')
+    f.write(json.dumps(destaddrs)+',\n')
+    f.write('"tx_list":')
+    f.write(json.dumps(tumbler.tx_list)+',\n')
+    f.write('"next_tx": 0,\n')
+    f.write('"pushed_tx": "",\n')
+    f.write('"pushed_destaddr": ""}\n')
+    f.close()
+
+def update_session_tx_confirmed (sessionname,i_tx):
+    sessionfilename = os.path.join(script_dir,'../sessions/'+sessionname)
+    if os.path.isfile(sessionfilename):
+        log.debug("update session tx confirmed "+sessionname+" "+str(i_tx))
+        with open(sessionfilename, "r") as sources:
+            lines = sources.readlines()
+            with open(sessionfilename, "w") as sources:
+                for line in lines:
+                    log.debug('line: '+line+'\n')
+                    if re.match(r'^["]next_tx["][:][ ][0-9]',line):
```

Comment:
General comment about this regexp matching approach: it should be far simpler to just json.loads() and then do the updates/changes purely in Python, then write back with json.dumps()